worker/src/main.rs

```diff
@@ -255,28 +296,24 @@ fn worker(
     // ------------------------------------------------------------------------
     // perform a remote attestation and get an unchecked extrinsic back
 
-    if skip_ra {
-        println!("[!] skipping remote attestation. will not register this enclave on chain");
-    } else {
-        // get enclaves's account nonce
-        let nonce = api.get_nonce_of(&tee_accountid).unwrap();
-        info!("Enclave nonce = {:?}", nonce);
-
-        let uxt =
-            enclave_perform_ra(eid, genesis_hash, nonce, config.ext_api_url.unwrap().as_bytes().to_vec()).unwrap();
+    // get enclaves's account nonce
+    let nonce = api.get_nonce_of(&tee_accountid).unwrap();
+    info!("Enclave nonce = {:?}", nonce);
 
-        let ue = UncheckedExtrinsic::decode(&mut uxt.as_slice()).unwrap();
+    let uxt = if skip_ra {
+        println!("[!] skipping remote attestation. Registering enclave without attestation report.");
+        enclave.mock_register_xt(api.genesis_hash, nonce, &config.ext_api_url.unwrap()).unwrap()
+    } else {
+        enclave_perform_ra(eid, genesis_hash, nonce, config.ext_api_url.unwrap().as_bytes().to_vec()).unwrap()
+    };
```

Comment:
My opinion is to prevent them wherever possible - usually. But here I deem it not very relevant, as this is while worker initialization. An error here basically means that the worker should not run; hence it is ok to panic. However, I think it is only ok in the current (suboptimal) state of the code.

When we refactor I would prefer to get rid of all of them and use `?`, or `map_err(|e| ..._)?` instead as it is also more ergonomic to read, but it needs a little extra work.

Reply:
Okay, so we'll accept them for now here - I can see your point about worker startup that can and should fail.
In PolkaDex we've taken inspiration from some of your code and also started using `.map_err()?;` quite extensively.